### ISSUE TITLE
Missing connections

### ISSUE THREAD

**Issue by SiriusAshling** [ori-rando/wotw-seedgen#24]
Some anchors assume the player comes from a certain direction, and omit "backwards" paths in that direction.
Since those anchors can be spawned on, this is a faulty assumption, and the connections will be logical gaps if the player spawns there.

The kii branch parser can detect anchors with no outgoing anchor connections. (This is only a subset of the problematic anchors).
Currently these are:
- HowlsDen.SecretRoom (since pre-rustgen)
- MidnightBurrows.LeverRoom (1da09c616a67d4bfd4688e5aeb4226ee7f7ad953)
- WestHollow.FarLeftRoom (since pre-rustgen)
- LowerWastes.ThirstyGorlek (654dac67829f333790d646c4b57be81d2f0fd302)

Scanning the logic visualizer would probably yield further missing connections.

For all of these either the missing connections should be implemented, or the anchor should be marked as nospawn.

**Comment by Foopyo** [comment]
These anchors do need more connections:
- WestHollow.HollowDrainLower needs a connection toward WestHollow.HollowDrainMiddle and one toward WestHollow.RockPuzzle (both would need to resolve first `WestHollow.FullyDrained` or include dirty swim in unsafe)
- InnerWellspring.PastSpinPuzzle needs a connection toward InnerWellspring.EastDoor. That one could be weird because it has to solve with `InnerWellspring.SpinPuzzleSolved` but that state won't be solved at the start of the seed so the state should probably be added from that anchor too.
- InnerWellspring.TopSecondRoom needs a connection to InnerWellspring.PastSpinPuzzle
- LowerReach.OutsideTPRoom needs a connection toward LowerReach.Teleporter
- LowerReach.ArenaArea paths are actually correct but the connection toward LowerReach.LowerWispPath should be toward LowerReach.WindSpinners instead
- UpperDepths.SecondKSRoom need a connection towards LowerDepths.East (which requires a ShurikenBreak). On top of that, there might be a need for an additional UpperDepths.SecondKSRoom <-> LowerDepths.Central connection if they aren't redundant with the UpperDepths.SecondKSRoom <-> LowerDepths.East connections
- LowerWastes.ThirstyGorlek needs a connection toward LowerWastes.TreeArea (and LowerWastes.TreeArea probably need to be renamed)
- LowerWastes.TreeArea connection toward LowerWastes.LeverArea need to go toward MuncherClimb instead
- UpperWastes.OutsideRuinsis needs a connection toward NorthTP

These anchors should be marked as no-spawn, at least in kii and under:
- WestHollow.HollowDrainMiddle since this anchor is initially under dirty water until the state `WestHollow.UpperDrainLeverPulled` is activated
- LowerReach.HalfwayIceCavern is an anchor used as a midpoint to make pathfinding easier. Even if we want to add backward compability (which could be used for a LowerReach.SwimmingPool->LowerReach.PastSnowball connection), this is not a really safe platform so it would be really bad for at least moki.

At least in kii, those anchor don't have any missing connections:
- InnerWellspring.ShortcutLever have connections to both InnerWellspring.EntranceDoor and InnerWellspring.WestDoor, which are the only possible connections
- UpperReach.OutsideTreeRoom is missing connections to some of the Frozenvariant of UpperReach.UpperSoup but since you need `UpperReach.ThawFurnace` in order to pass the waterfall, that anchor isn't accessible from here. the UpperReach.UpperSoup  connection should probably be connecting to UpperReach.UpperSoupThawed instead.
- UpperDepths.BelowHive seems to be connected to everything beside maybe the Mora path, if there is some ShurikenBreak that wouldn't be redundant with paths from the teleporter
- PoolsApproach.LeftOfWheel have connections to PoolsApproach.RightOfWheel, EastPools.LeverRoom. The connection toward MarshSpawn.Main need to go toward MarshSpawn.PoolsPath instead.

**Comment by SiriusAshling** [comment]
Just adding a note since it doesn't seem to be present here yet: A path to the ShriekTP would also be relevant in unsafe because it could give the NonGladesTeleporter state

**Comment by Foopyo** [comment]
> Just adding a note since it doesn't seem to be present here yet: A path to the ShriekTP would also be relevant in unsafe because it could give the NonGladesTeleporter state

The Shriek TP path is a pretty massive one to pathfind for because you need to destroy all willow's hearts and these have been disreguarded since gorlek's pathfinding

**Comment by Foopyo** [comment]
It can also be noted that if you can get logical access to all willow's hearts, you can also access WillowsEnd.InnerTP, which would be a less restrictive way to get the NonGladesTeleporter state. So this will only be useful when using headers which can grant the Shriek teleporter pickup or when playing with No willows Hearts.

**Comment by SiriusAshling** [comment]
Yeah No Willow Hearts + Spawn in front of the Shriek door + Spawn with Glades TP would be a simple example

**Comment by Foopyo** [comment]
Doesn't random spawn solves the issue? You'll get a point on your map on which you tp to, so you need the teleporter in order to get the pickup.

**Comment by Foopyo** [comment]
logic could probably just set that state to true if you spawn somewhere else than at glades TP

**Comment by Foopyo** [comment]
Closing the issue since all the pathfinding has been done. All the paths validation can get done during the kii validation process, except for MidnightBurrows for which the issue #107 got created.
